src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -715,4 +715,13 @@ public boolean hasIndicatedSections(String courseId) throws EntityDoesNotExistEx
         return archivedCourseIds;
     }
 
+    /**
+     * Returns all courses in the Datastore.
+     * @deprecated Not scalable. Use only in admin features.
+     */
+    @Deprecated
+    public List<CourseAttributes> getAllCourses() {
```

Comment:
Yes, it can be done ([`StatisticsPerInstitute.java`](https://github.com/TEAMMATES/teammates/pull/7672/files#diff-bf22c59f8430606606f182e63b7ecb52) from my latest PR should run fine); just that the client script code would look uglier and can become outdated if the schema changes (as opposed to production code which would be kept updated if the schema were to change). However, I feel that this is fine as it is in the nature of client scripts.